lib/workers/repository/finalise/repository-statistics.ts

```diff
@@ -33,3 +35,55 @@ export function runRenovateRepoStats(
   }
   logger.debug({ stats: prStats }, `Renovate repository PR statistics`);
 }
+
+interface BranchMetadata {
+  branchName: string;
+  sha: string | null;
+  parentSha: string | null;
+  automerge: boolean;
+  isModified: boolean;
+}
+
+interface BaseBranchMetadata {
+  branchName: string;
+  sha: string;
+}
+
+export function runBranchSummery(): void {
+  const { scan, branches } = getCache();
+
+  const baseMetadata: BaseBranchMetadata[] = [];
+  for (const [branchName, cached] of Object.entries(scan ?? {})) {
+    baseMetadata.push({ branchName, sha: cached.sha });
+  }
+
+  function unwrap({
```

Comment:
is there a reason why this function is nested? can u get it outside? its confusing to readers (to me at least)

Reply:
it is nested as it is only used within its scope. usually i would make in an inline arrow function, but it is too long even though its only one statement.

Reply:
this way it is also closer to where it is being used

Reply:
don't use nested functions, they cause reevaluate on each outer function call. so move to outer scope, makes code also more readable

Reply:
https://github.com/renovatebot/renovate/pull/17065